src/Chapter3.hs

```diff
@@ -971,6 +1115,26 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday deriving (Show, Eq, Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend day = any ((== True) . (== day)) [Saturday,Sunday]
+
+nextDay :: WeekDay -> WeekDay
+nextDay Sunday = Monday
+nextDay d = succ d
+
+daysToParty :: WeekDay -> Int
+daysToParty Friday = 0
+daysToParty d = 1 + daysToParty (nextDay d)
```

Comment:
Your implementation is correct :+1: And nice way of using already implemented `nextDay`!

Though, this implementation can be a bit more efficient, if you use the `fromEnum` method of `Enum` typeclass